src/core/utils.ts

```diff
@@ -355,6 +355,10 @@ export function shallowEqualObjects<T>(a: T, b: T): boolean {
   return true
 }
 
+export function isPlainArray(a: any) {
```

Comment:
a can probably be of type `unknown`. Also, I would appreciate it if we rename it to something like `value`:

```
export function isPlainArray(value: unknown) {